pkg/client/cache/listers.go

```diff
@@ -111,6 +111,44 @@ func (s storePodsNamespacer) List(selector labels.Selector) (api.PodList, error)
 	return pods, nil
 }
 
+// Please note that selector is filtering among the pods that have gotten into
+// the store; there may have been some filtering that already happened before
+// that.
+func (s storePodsNamespacer) ListWithPredicate(pickPod func(*api.Pod) bool) (api.PodList, error) {
```

Comment:
> @deads2k

Eh, I'd do it differently.  I just looked at the title and I'm assuming you're doing this to run with a SharedInformer for GC?  If that's the case, I would have created an index based on the label/field selector that was previously used for the GC watch, then queried my index.

I'm fine with the API in general, though I'd say you should have the List delegate here with a wrapper on its label selector.